document-store/src/integrationTest/java/org/hypertrace/core/documentstore/mongo/MongoDocStoreTest.java

```diff
@@ -64,6 +64,32 @@ public void cleanup() {
     datastore.createCollection(COLLECTION_NAME, null);
   }
 
+  @Test
+  public void testOffsetAndLimit() throws IOException {
+    Collection collection = datastore.getCollection(COLLECTION_NAME);
+    collection.upsert(new SingleValueKey("default", "testKey1"), createDocument("foo1", "bar1"));
+    collection.upsert(new SingleValueKey("default", "testKey2"), createDocument("foo2", "bar2"));
+    collection.upsert(new SingleValueKey("default", "testKey3"), createDocument("foo3", "bar3"));
+    collection.upsert(new SingleValueKey("default", "testKey4"), createDocument("foo4", "bar4"));
+    collection.upsert(new SingleValueKey("default", "testKey5"), createDocument("foo5", "bar5"));
+
+    Query query = new Query();
+    query.setLimit(2);
+    query.setOffset(1);
+
+    Iterator<Document> results = collection.search(query);
```

Comment:
Can you extend this test to the same query twice with pagination? That will cover the default order test.